contracts/contracts/opl/Endpoint.sol

```diff
@@ -0,0 +1,239 @@
+        if (epExists) result = endpoints[epSel](message);
+        // Convert the Result to a Celer ExecutionStatus.
+        if (result == Result.TransientFailure) return 2; // ExecutionStatus.Retry
+        if (result == Result.Success) return 1; // ExecutionStatus.Success
+        return 0; // ExecutionStatus.Fail
+    }
+
+    function estimateFee(bytes memory _message)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 feeBase = ICelerMessageBus(bus).feeBase();
+        uint256 feePerByte = ICelerMessageBus(bus).feePerByte();
+        return
+            feeBase +
+            (_message.length +
+                32 + /* seq */
+                4) * /* epsel */
+            feePerByte;
+    }
+
+    function _chainName2ChainId(bytes32 name) internal pure returns (uint256) {
+        if (name == "ethereum") return 1;
+        if (name == "goerli") return 5;
+        if (name == "optimism") return 10;
+        if (name == "bsc") return 56;
+        if (name == "bsc-testnet") return 97;
+        if (name == "polygon") return 137;
+        if (name == "fantom") return 0xfa;
+        if (name == "fantom-testnet") return 0xfa2;
+        if (name == "moonriver") return 0x505;
+        if (name == "sapphire-testnet") return 0x5aff;
+        if (name == "arbitrum-one") return 1;
+        if (name == "arbitrum-nova") return 1;
+        if (name == "sapphire") return 0x5afe;
+        if (name == "polygon-mumbai") return 80001;
+        if (name == "avalanche") return 43114;
+        if (name == "avalanche-fuji") return 43313;
+        if (name == "arbitrum-testnet") return 0x66eeb;
+        return 0;
+    }
+
+    function _getChainConfig(uint256 _chainId)
+        internal
+        pure
+        returns (address _bus, bool _isTestnet)
+    {
+        if (_chainId == 1)
+            // ethereum
+            return (0x4066D196A423b2b3B8B054f4F40efB47a74E200C, false);
+        if (_chainId == 5)
+            // goerli
+            return (0xF25170F86E4291a99a9A560032Fe9948b8BcFBB2, true);
+        if (_chainId == 10)
+            // optimism
+            return (0x0D71D18126E03646eb09FEc929e2ae87b7CAE69d, false);
+        if (_chainId == 56)
+            // bsc
+            return (0x95714818fdd7a5454F73Da9c777B3ee6EbAEEa6B, false);
+        if (_chainId == 97)
+            // bsc testnet
+            return (0xAd204986D6cB67A5Bc76a3CB8974823F43Cb9AAA, true);
+        if (_chainId == 137)
+            // polygon
+            return (0xaFDb9C40C7144022811F034EE07Ce2E110093fe6, false);
+        if (_chainId == 0xfa)
+            // fantom
+            return (0xFF4E183a0Ceb4Fa98E63BbF8077B929c8E5A2bA4, false);
+        if (_chainId == 0xfa2)
+            // fantom testnet
+            return (0xb92d6933A024bcca9A21669a480C236Cbc973110, true);
+        if (_chainId == 0x505)
+            // moonriver
+            return (0x940dAAbA3F713abFabD79CdD991466fe698CBe54, false);
+        if (_chainId == 0x5aff)
```

Comment:
No Mainnet yet?

Reply:
Celer said they'd be deploying within a week.